Source/IGListAdapter.m

```diff
@@ -534,15 +534,26 @@ - (void)updateBackgroundViewShouldHide:(BOOL)shouldHide {
     if (self.isInUpdateBlock) {
         return; // will be called again when update block completes
     }
-    UIView *backgroundView = [self.dataSource emptyViewForListAdapter:self];
-    // don't do anything if the client is using the same view
-    if (backgroundView != _collectionView.backgroundView) {
-        // collection view will just stack the background views underneath each other if we do not remove the previous
-        // one first. also fine if it is nil
-        [_collectionView.backgroundView removeFromSuperview];
-        _collectionView.backgroundView = backgroundView;
+
+    if (shouldHide) {
+        _emptyListView.hidden = YES;
+    } else {
+        UIView *const newBackgroundView = [self.dataSource emptyViewForListAdapter:self];
```

Comment:
nit: remove `const`, been keeping those just for primitives since you can still mutate the object